csm_web/scheduler/views/section.py

```diff
@@ -779,3 +784,198 @@ def wotd(self, request, pk=None):
+                    receiver: [`List of Swap objects`],
+                    sender: [`List of Swap objects`]
+                }
+                Where each Swap object has the following format:
+                {
+                    swap_id: int,
+                    name: string,
+                    mentor_name: string,
+                    time: DateTime,
+                }
+
+        POST: Handles creating a new Swap instance when a Swap a requested or
+        processes an existing Swap request.
+            Request format:
+                { email: string }
+                `email` is the email of the student who will be receiving the swap request.
+            Response format:
+                {}
+                With status code of 201 if successful, 400 if not.
+        """
+        section = get_object_or_error(Section.objects, pk=section_id)
+        student_id = request.data.get("student_id", -1) if request.method != "GET" \
+            else request.query_params.get("student_id", -1)
+
+        # check if student_id is specified in request
+        if student_id == -1:
+            raise PermissionDenied("The `student_id` field in the request cannot be empty,"
+                                   "please specify student.")
+
+        # check if student exists
+        try:
+            student = get_object_or_error(Student.objects, id=student_id)
+        except Exception:
+            raise NotFound("No student found with id: " + str(student_id))
+        if request.method == "GET":
+            if section.mentor.user == request.user:
+                raise PermissionDenied("Cannot `GET` swaps for a mentor.")
+            try:
+                outgoing_swaps = Swap.objects.filter(sender=student)
+                incoming_swaps = Swap.objects.filter(receiver=student)
+            except Exception as e:
+                raise NotFound("No swaps found for student with id: " + str(student_id))
+            
+            student_swaps = {
+                "sender": SwapSerializer(outgoing_swaps, many=True).data,
+                "receiver": SwapSerializer(incoming_swaps, many=True).data,
+            }
+            return Response(student_swaps, status=status.HTTP_200_OK)
+
+        if request.method == "POST":
+            if section.mentor.user == request.user:
+                logger.error("Cannot `POST` swaps for a mentor.")
+                raise PermissionDenied("Cannot `POST` swaps for a mentor.")
+
+            '''Create a new Swap object between the the current student
+            and the student with the email specified in the request. This is essentially
+            creating a swap invite.'''
+            receiver_email = request.data.get("receiver_email", None)
+
+            # Check if receiver_email is empty
+            if receiver_email is None:
+                raise PermissionDenied("The `receiver_email` field in the request cannot be empty,"
+                                        "please specify a receiver.")
```

Comment:
User input errors should be returned as a 400 Bad Request error, not a 403 Permission Denied error (unsure if there's a builtin response class for this). A 403 (raised by `PermissionDenied`) is for when the user is not authorized to access a resource or perform an action.